server/app/services/applicant/ApplicantData.java

```diff
@@ -66,84 +76,185 @@ public void setPreferredLocale(Locale locale) {
-    }
-    if (lastName != null) {
-      putString(WellKnownPaths.APPLICANT_LAST_NAME, lastName);
-    }
+    return applicant.getLastName().or(() -> readString(WellKnownPaths.APPLICANT_LAST_NAME));
   }
 
   public Optional<String> getPhoneNumber() {
-    return readString(WellKnownPaths.APPLICANT_PHONE_NUMBER);
+    return applicant.getPhoneNumber().or(() -> readString(WellKnownPaths.APPLICANT_PHONE_NUMBER));
   }
 
   public void setPhoneNumber(String phoneNumber) {
+    applicant.setPhoneNumber(phoneNumber);
     putPhoneNumber(WellKnownPaths.APPLICANT_PHONE_NUMBER, phoneNumber);
   }
 
-  public void setUserName(String displayName) {
+  public Optional<LocalDate> getDateOfBirth() {
+    return applicant
+        .getDateOfBirth()
+        .or(
+            () -> {
+              Path dobPath = WellKnownPaths.APPLICANT_DOB;
+              Path deprecatedDobPath = WellKnownPaths.APPLICANT_DOB_DEPRECATED;
+              return hasPath(dobPath) ? readDate(dobPath) : readDate(deprecatedDobPath);
+            });
+  }
+
+  public void setDateOfBirth(String dateOfBirth) {
+    applicant.setDateOfBirth(dateOfBirth);
+
+    // The new path is underneath the old path, so we have to
+    // specifically check the old path exists but not the new one.
+    Path newPath = WellKnownPaths.APPLICANT_DOB;
+    Path oldPath = WellKnownPaths.APPLICANT_DOB_DEPRECATED;
+    Path dobPath = hasPath(oldPath) && !hasPath(newPath) ? oldPath : newPath;
+    putDate(dobPath, dateOfBirth);
+  }
+
+  /**
+   * Parses a name string to extract the first, middle, and last names, if they exists, and sets
+   * those fields.
+   *
+   * @param displayName A string that contains the applicant's name, with first, middle, and last
+   *     separated by spaces. May provide only first name or only first last.
+   * @param overwrite Overwrite any existing data.
+   */
+  public void setUserName(String displayName, boolean overwrite) {
     String firstName;
-    String lastName = null;
-    String middleName = null;
+    Optional<String> lastName = Optional.empty();
+    Optional<String> middleName = Optional.empty();
     List<String> listSplit = Splitter.on(' ').splitToList(displayName);
     switch (listSplit.size()) {
       case 2:
         firstName = listSplit.get(0);
-        lastName = listSplit.get(1);
+        lastName = Optional.of(listSplit.get(1));
         break;
       case 3:
         firstName = listSplit.get(0);
-        middleName = listSplit.get(1);
-        lastName = listSplit.get(2);
+        middleName = Optional.of(listSplit.get(1));
+        lastName = Optional.of(listSplit.get(2));
         break;
       case 1:
         // fallthrough
       default:
         // Too many names - put them all in first name.
         firstName = displayName;
     }
-    setUserName(firstName, middleName, lastName);
+    setUserName(firstName, middleName, lastName, overwrite);
   }
 
+  public void setUserName(String displayName) {
```

Comment:
comment (non-blocking): I realized this method is only used in tests/test setup. We could refactor those calls to use the one that looks for first middle last so we could get rid of one of the `setUserName` functions. I don't think you need to do that in this PR, but I might look at that when I'm doing system actions if that makes sense to you.